pkg/datamover/backup_micro_service.go

```diff
@@ -127,15 +127,13 @@ func (r *BackupMicroService) Init() error {
 	return err
 }
 
-var waitControllerTimeout time.Duration = time.Minute * 2
-
 func (r *BackupMicroService) RunCancelableDataPath(ctx context.Context) (string, error) {
 	log := r.logger.WithFields(logrus.Fields{
 		"dataupload": r.dataUploadName,
 	})
 
 	du := &velerov2alpha1api.DataUpload{}
-	err := wait.PollUntilContextTimeout(ctx, 500*time.Millisecond, waitControllerTimeout, true, func(ctx context.Context) (bool, error) {
+	err := wait.PollUntilContextCancel(ctx, 500*time.Millisecond, true, func(ctx context.Context) (bool, error) {
```

Comment:
We cannot use Get:
This is executed inside the data mover pod
The data mover pod needs to hold its execution until the controller in the node-agent turns the DU/DD to InProgress. Otherwise, there will be some synchronization problem.

Reply:
After a further thinking, I think an infinite wait is better:

- The controller is the part that controls the workflow, so the data mover pods just need to action on calls of the controller
- The controller doesn't want to see a "surprise" like the data mover pods exit just because it didn't change the DU/DD to InProgress in time. This makes the control logic/troubleshooting more complex
- The data mover pods won't fall into a dead wait even there is no timeout here, the controller will delete the pods whenever an error happens without worrying about that the VGDP already starts to transfer data, as long as the DU/DD does not go to InProgress

Reply:
Agree with the change, 3rd point helps clear things up.